presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -656,6 +663,27 @@ void PrestoServer::run() {
   }
 }
 
+void PrestoServer::startLinuxMemoryChecker() {
+#ifdef PRESTO_ENABLE_LINUX_MEMORY_CHECKER
+  uint64_t systemMemLimitBytes = systemConfig->systemMemLimitGb();
+  uint64_t systemMemShrinkBytes = systemConfig->systemMemShrinkGb();
+  linuxMemoryChecker_ =
+      std::make_unique<LinuxMemoryChecker>(PeriodicMemoryChecker::Config{
```

Comment:
Can we change this? We don't need to provide all config options. Most of these are the default to something like
```
PeriodicMemoryChecker::Config config;
config.systemMemPushbackEnabled = systemConfig->systemMemPushbackEnabled();
config.systemMemLimitBytes = systemConfig->systemMemLimitGb() << 30;
config.systemMemShrinkBytes = systemConfig->systemMemShrinkGb() << 30;
linuxMemoryChecker_ = std::make_unique<LinuxMemoryChecker>(config);
```

The Memory checker doesn't use any of the other options anyway. So we can use the defaults in the Config.

And remove lines 668 and 669 as well if this works ok?

Reply:
Changing to this:
```
PeriodicMemoryChecker::Config config;
  config.systemMemPushbackEnabled = systemConfig->systemMemPushbackEnabled();
  config.systemMemLimitBytes = systemConfig->systemMemLimitGb() << 30;
  config.systemMemShrinkBytes = systemConfig->systemMemShrinkGb() << 30;
  config.mallocMemHeapDumpEnabled = systemConfig->mallocMemHeapDumpEnabled();
  config.minHeapDumpIntervalSec = systemConfig->mallocMemMinHeapDumpInterval();
  config.maxHeapDumpFiles = systemConfig->mallocMemMaxHeapDumpFiles();
  linuxMemoryChecker_ = std::make_unique<LinuxMemoryChecker>(config);
  ```

Reply:
But the Linux Memory Checker is not using these options
```  
  config.mallocMemHeapDumpEnabled = systemConfig->mallocMemHeapDumpEnabled();
  config.minHeapDumpIntervalSec = systemConfig->mallocMemMinHeapDumpInterval();
  config.maxHeapDumpFiles = systemConfig->mallocMemMaxHeapDumpFiles();
```

Setting them implies that they can be used and any value change affects the `LinuxMemoryChecker`.

Reply:
Ahh I see, removing those 3 options